independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -28,56 +30,84 @@ public class InterceptorInfo extends BeanInfo implements Comparable<InterceptorI
 
     private final Set<AnnotationInstance> bindings;
 
-    private final MethodInfo aroundInvoke;
+    private final List<MethodInfo> aroundInvokes;
 
-    private final MethodInfo aroundConstruct;
+    private final List<MethodInfo> aroundConstructs;
 
-    private final MethodInfo postConstruct;
+    private final List<MethodInfo> postConstructs;
+
+    private final List<MethodInfo> preDestroys;
 
-    private final MethodInfo preDestroy;
 
     InterceptorInfo(AnnotationTarget target, BeanDeployment beanDeployment, Set<AnnotationInstance> bindings,
             List<Injection> injections, int priority) {
         super(target, beanDeployment, BuiltinScope.DEPENDENT.getInfo(),
                 Collections.singleton(Type.create(target.asClass().name(), Kind.CLASS)), new HashSet<>(), injections,
                 null, null, false, Collections.emptyList(), null, false, null, priority);
         this.bindings = bindings;
-        List<MethodInfo> aroundInvokes = new ArrayList<>();
-        List<MethodInfo> aroundConstructs = new ArrayList<>();
-        List<MethodInfo> postConstructs = new ArrayList<>();
-        List<MethodInfo> preDestroys = new ArrayList<>();
+        aroundInvokes = new ArrayList<>();
+        aroundConstructs = new ArrayList<>();
+        postConstructs = new ArrayList<>();
+        preDestroys = new ArrayList<>();
 
         ClassInfo aClass = target.asClass();
-        while (aClass != null) {
-            for (MethodInfo method : aClass.methods()) {
-                if (Modifier.isStatic(method.flags())) {
-                    continue;
-                }
-                if (method.hasAnnotation(DotNames.AROUND_INVOKE)) {
-                    aroundInvokes.add(validateSignature(method));
-                }
-                if (method.hasAnnotation(DotNames.AROUND_CONSTRUCT)) {
-                    aroundConstructs.add(validateSignature(method));
-                }
-                if (method.hasAnnotation(DotNames.POST_CONSTRUCT)) {
-                    postConstructs.add(validateSignature(method));
-                }
-                if (method.hasAnnotation(DotNames.PRE_DESTROY)) {
-                    preDestroys.add(validateSignature(method));
-                }
+        processInheritedMethods(aClass, beanDeployment);
+
+        if (aroundConstructs.isEmpty() && aroundInvokes.isEmpty() && preDestroys.isEmpty() && postConstructs.isEmpty()) {
+            LOGGER.warnf("%s declares no around-invoke method nor a lifecycle callback!", this);
+        }
+    }
+
+    private void processInheritedMethods(ClassInfo aClass, BeanDeployment beanDeployment) {
+        DotName superTypeName = aClass.superName();
+        if (superTypeName != null && !DotNames.OBJECT.equals(superTypeName)) {
+            ClassInfo superClassInfo = getClassByName(beanDeployment.getBeanArchiveIndex(), superTypeName);
+            if (superClassInfo == null) {
+                throw new DefinitionException(
```

Comment:
I'm not sure whether this should result in a `DefinitionException`. Also previously it was just ignored. But it should only happen rarely. @manovotn @Ladicek WDYT?

Reply:
So currently we just silently ignore it? I'd say an exception is good since otherwise this will affect functionality in unexpected ways. But `DefinitionException` is CDI specific and we should use some other exception, for instance illegal state sounds fine to me.

Reply:
Agree this is not a definition error. This really should never happen in Quarkus -- if it does, we have an incomplete classpath. A `RuntimeException` (or `IllegalStateException`) sounds more appropriate to me too.